src/app/api/mobile/courses/[courseId]/[collectionId]/[contentId]/route.ts

```diff
@@ -0,0 +1,62 @@
+import db from '@/db';
+import { NextRequest, NextResponse } from 'next/server';
+
+async function checkUserContentAccess(userId: string, contentId: string) {
+  const userContent = await db.content.findFirst({
+    where: {
+      id: parseInt(contentId, 10),
+      courses: {
+        some: {
+          course: {
+            purchasedBy: {
+              some: {
+                userId: userId,
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+  return userContent !== null;
+}
+
+export async function GET(req : NextRequest,{ params }: { params: { contentId: string } }) {
+  try {
+    const { contentId } = params;
+    const user = JSON.parse(req.headers.get('g') || '');
+    const userContentAccess = await checkUserContentAccess(user.id, contentId);
+    if (!userContentAccess) {
+      return NextResponse.json({ message: 'User does not have access to this content' }, { status: 403 });
+    }
+    const contents = await db.content.findUnique({
```

Comment:
We are already making a db call for finding if the user has the course access or not. Why don't we reuse the query and get the contents from there itself?

https://github.com/code100x/cms/blob/45668f675f430bb0eb6c8490f6a9d76bd397c997/src/app/api/mobile/courses/%5BcourseId%5D/%5BcollectionId%5D/%5BcontentId%5D/route.ts#L4-L22